core/sql/common/charinfo.h

```diff
@@ -260,11 +260,14 @@ class CharInfo
                                          Int32   sourceLenInBytes,
                                          CharSet targetCS);
 
+  static void initBuiltinCollationDB();
+ 
+
 private:
 friend class CollationDB;			// needs to access builtinCDB_
 
    static const char*	const localeCharSet_;
-   static const CollationDB   builtinCollationDB_;
+   static const CollationDB   *builtinCollationDB_;
```

Comment:
Suggest initializing to NULL. So if you get a thread race condition, you'll get better behavior if one thread accessing this pointer beats the other thread that is initializing it.

Reply:
Oh, sorry; this is the declaration rather than the definition. Ignore my comment.